modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/PredicateTokenFilterScriptFactory.java

```diff
@@ -69,5 +69,11 @@ protected boolean accept() throws IOException {
             token.updatePosition();
             return script.execute(token);
         }
+
+        @Override
+        public void reset() throws IOException {
```

Comment:
I think `ScriptedConditionTokenFilterFactory` has the same issue ?

Reply:
++, will update and add a test